src/AssetsListener.php

```diff
@@ -0,0 +1,367 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace Zend\Mvc;
+
+use Zend\EventManager\AbstractListenerAggregate;
+use Zend\EventManager\EventManagerInterface;
+use Zend\Mvc\MvcEvent;
+use Zend\Stdlib\ResponseInterface;
+use Zend\Http\Response\Stream as StreamResponse;
+use Zend\View\Assets\Asset;
+use Zend\View\Assets\AssetsManager;
+use Zend\Router\RouteInterface;
+use Zend\View\Resolver\ResolverInterface;
+use Zend\ServiceManager\PluginManagerInterface;
+
+/**
+ * Description of DispatchAssetsListener
+ *
+ * @author Shiri
+ */
+class AssetsListener extends AbstractListenerAggregate
+{
+    /**
+     * @var AssetsManager
+     */
+    protected $assetsManager;
+
+    /**
+     * @var ResolverInterface
+     */
+    protected $assetsResolver;
+
+    /**
+     * @var PluginManagerInterface
+     */
+    protected $filterManager;
+
+    protected $request;
+
+    protected $routeName = 'assets';
+
+    protected $routerCacheFolder = 'assets';
+
+    protected $useInternalRouter = true;
+
+    protected $router;
+
+    protected $isCacheToPublic = true;
+
+    public function attach(EventManagerInterface $events, $priority = 1)
+    {
+        $this->listeners[] = $events->attach(MvcEvent::EVENT_DISPATCH, array($this, 'onDispatch'), 20);
+        $this->listeners[] = $events->attach(MvcEvent::EVENT_BOOTSTRAP, array($this, 'onBootstrap'), -1000);
+    }
+
+    public function onBootstrap(MvcEvent $event)
+    {
+        if (!$this->useInternalRouter) {
+            return;
+        }
+
+        if ($this->router) {
+            $router = $this->router;
+        } else {
+            $router = [
+                'type' => \Zend\Router\Http\Segment::class,
+                'options' => [
+                    'route' => '/' . $this->routerCacheFolder . '[/alias-:alias][/prefix-:prefix]/:asset',
```

Comment:
Wonder if this could do with some validation? e.g. make sure it adheres to `Segment` syntax etc.